browser/brave_rewards/android/brave_rewards_native_worker.cc

```diff
@@ -109,6 +112,36 @@ bool BraveRewardsNativeWorker::IsRewardsEnabled(JNIEnv* env) {
       ->GetBoolean(brave_rewards::prefs::kEnabled);
 }
 
+bool BraveRewardsNativeWorker::ShouldShowSelfCustodyInvite(JNIEnv* env) {
+  bool isSelfCustodyInviteDismissed =
+      ProfileManager::GetActiveUserProfile()
+          ->GetOriginalProfile()
+          ->GetPrefs()
+          ->GetBoolean(brave_rewards::prefs::kSelfCustodyInviteDismissed);
+  std::string country_code = brave_rewards_service_->GetCountryCode();
+  const std::vector<std::string> providers =
+      brave_rewards_service_->GetExternalWalletProviders();
+  if (!base::Contains(providers,
+                      brave_rewards::internal::constant::kWalletSolana)) {
+    return false;
+  }
+
+  const auto& regions = parameters_->wallet_provider_regions.at(
+      brave_rewards::internal::constant::kWalletSolana);
+  if (!regions) {
+    return false;
```

Comment:
I'm thinking this should return `true` here: if the "solana" regions information is missing for some reason, then still let the user see the invitation and attempt to connect.

Reply:
if the regions information is missing, would it be safe to allow users to connect with solana ?

Reply:
@Miyayes mentioned that we should be good as we have other restrictions on backend. I am updating the change.